parser/src/config/mod.rs

```diff
@@ -0,0 +1,400 @@
+    }
+}
+impl fmt::Display for Format {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(self.id())
+    }
+}
+
+impl std::str::FromStr for Format {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        for format in Format::all() {
+            if format.id() == s {
+                return Ok(*format);
+            }
+        }
+        Err(format!("invalid format id: '{}'", s))
+    }
+}
+
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize, schemars::JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct ParseConfig {
+    /// format forces the use of the given parser and disables automatic format detection. If
+    /// unspecified, then the format will be inferred from the filename, content-type, or file
+    /// contents.
+    #[serde(default)]
+    pub format: Option<Format>,
+
+    /// filename is used for format inference. It will be ignored if `format` is specified.
+    #[serde(default)]
+    pub filename: Option<String>,
+
+    /// The mime type of the data, if known. This will be used for format inference, or ignored if
+    /// `format` is specified.
+    #[serde(default)]
+    pub content_type: Option<String>,
+
+    #[serde(default)]
+    /// Add the record offset as a property of each object at the location given. The offset is a
+    /// monotonic counter that starts at 0 and increases by 1 for each output document.
+    pub add_record_offset: Option<JsonPointer>,
+
+    /// Static data to add to each output JSON document.
+    #[serde(default)]
+    pub add_values: BTreeMap<JsonPointer, Value>,
+
+    /// Projections control how tabular data like CSV gets transformed into potentially nested JSON
+    /// structures. The keys are field names, which may match column names in the source data, and
+    /// the values are json pointers indicating where to place the values within the output JSON
+    /// document.
+    #[serde(default)]
+    pub projections: BTreeMap<String, JsonPointer>,
+
+    /// JSON schema describing the desired shape of the output JSON documents. Output documents
+    /// will not be validated against this schema, but it can be used to automatically infer
+    /// projections for mapping tabular data to nested JSON structures.
+    #[serde(default)]
+    pub schema: Value,
+
+    /// Mappings from file extensions to format identifiers.
+    #[serde(default)]
+    pub file_extension_mappings: BTreeMap<String, Format>,
+
+    /// Mappings from content types to format identifiers.
+    #[serde(default)]
+    pub content_type_mappings: BTreeMap<String, Format>,
+
+    // Below are format-specific configurations, which are used by the parsers.
+    /// Configures handling of comma-separated values (CSV) format.
+    #[serde(default)]
+    pub csv: Option<CharacterSeparatedConfig>,
+
+    /// Configures handling of tab-separated values (TSV) format.
+    #[serde(default)]
+    pub tsv: Option<CharacterSeparatedConfig>,
```

Comment:
Ah, I think I forgot to respond to this one earlier. The intent of making tsv a separate format is so that you could import both types of files from the same capture. So you could have your TSV files use a different config than your CSVs. Also just to more or less correspond with the file extensions that people use.